server/container/util/src/main/java/org/apache/james/util/scanner/SpamAssassinInvoker.java

```diff
@@ -202,17 +184,25 @@ public boolean learnAsSpam(InputStream message, String user) throws MessagingExc
      *             if an error occured during learning.
      */
     public boolean learnAsHam(InputStream message, String user) throws MessagingException {
+        return metricFactory.withMetric(
+            "spamAssassin-ham-report",
+            Throwing.supplier(
+                () -> performReport(message, user, "ham"))
+                .sneakyThrow());
+    }
+
+    public boolean performReport(InputStream message, String user, String messageClass) throws MessagingException {
```

Comment:
This method name is not really clear, IMO. Maybe `learn`

and also it should be private.

Reply:
I prefer matthieu **reportMessageAs** proposal.